core/src/container.cpp

```diff
@@ -81,7 +81,7 @@ bool ContainerBasePrivate::traverseStages(const ContainerBase::StageCallback& pr
 
 	for (auto& stage : children_) {
 		if (!processor(*stage, cur_depth))
-			continue;
+			return false;
```

Comment:
I had to change the semantics of `traverseStages()`:
So far, returning false from the processor function, just skipped further traversing the current child (depth-wise).
Now, traversal is completely aborted, even not traversing the remaining siblings of the current child.
@v4hn: Having a single boolean return value, we cannot distinguish these two cases...

Reply:
Strictly speaking, you do not need this change, but it avoids more book-keeping in the lambdas.

Looking at the history of this, it looks like you arbitrarily defined the current semantics a long way back.
Do you have a use-case for the current semantics (continue with siblings) in mind?
I would have preferred abort-on-false to begin with.

This change deserves its own explanation in a commit though.

Reply:
> Strictly speaking, you do not need this change, but it avoids more book-keeping in the lambdas.

I'm curious how you would solve this without the change. I guess, you wouldn't care about traversing the whole list in all cases, even if a match was already found.

Separated the traversal change into its own commit.

Reply:
> I'm curious how you would solve this without the change

If we really wouldn't want to change semantics, you could just add static boolean guards instead of returning false, but we both agree that the API is not that fixed yet.

using the traversal operator to count instead of adding (previously private) API seems like a sledgehammer solution anyway, but sure, it works.

Thank you for resolving this :+1: